websites/O/Oculus/presence.ts

```diff
@@ -0,0 +1,208 @@
+                  presenceData.state = `Showcase: ${showCase ?? "Loading..."}`;
+                  presenceData.buttons.push({
+                    label: "View showcase",
+                    url: `https://${hostName}/${path[0]}/${path[1]}/${path[2]}/${path[3]}`
+                  });
+
+                  // Developer posts
+              } else if (path[2] === "developer-post") {
+                  const title = document.querySelector("_9cq4")?.textContent ?? "Unknown dev-post";
+                  presenceData.state = `Dev-post: ${title.length > 118 ? `${title.slice(0, 115)}...` : title}`;
+                  presenceData.buttons.push({
+                    label: "Read dev-post",
+                    url: `https://${hostName}/${path[0]}/${path[1]}/${path[2]}/${path[3]}`
+                  });
+
+                  // Searching
+              } else if (path[2] === "search") {
+                  const searchText = document.querySelector("disco-search__query")?.textContent;
+                  presenceData.state = `Searching for: ${searchText}`;
+                  presenceData.smallImageKey = "search";
+
+                  // Bundles
+              } else if (document.querySelector("div.bundle-detail-page__description > h1")?.textContent) {
+                  const bundle = document.querySelector("div.bundle-detail-page__description > h1")?.textContent;
+                  presenceData.state = `Bundle: ${bundle ?? "Loading..."}`;
+                  presenceData.buttons.push({
+                    label: "View bundle",
+                    url: `https://${hostName}/${path[0]}/${path[1]}/${path[2]}`
+                  });
+
+                  // Games
+              } else {
+                  const game = document.getElementsByClassName("app-description__title")[0]?.textContent;
+                  presenceData.state = `Game: ${game ?? "Loading..."}`;
+                  presenceData.buttons.push({
+                    label: "View game",
+                    url: `https://${hostName}/${path[0]}/${path[1]}/${path[2]}`
+                  });
+
+              }
+              break;
+            }
+            case "research": {
+              presenceData.details = "Viewing the page:";
+              presenceData.state = "Research";
+              break;
+            }
+            case "careers": {
+              presenceData.details = "Viewing the page:";
+              presenceData.state = "Career";
+              break;
+            }
+            default: {
+              presenceData.details = "Viewing the page:";
+              presenceData.state = "Unknown page";
+              break;
+            }
+          }
+      }
+      break;
+    }
+  }
+
+  if (!presenceData.details) {
+    presence.setTrayTitle();
+    presence.setActivity();
+  } else {
+    // Delete button(s) / timestamp relating to the setting
+    if (presenceData.buttons?.length && !showButtons) delete presenceData.buttons;
```

Comment:
```suggestion
    if (presenceData.buttons?.length === 0 && !showButtons) delete presenceData.buttons;
```
SlowLife removed it idk why

Reply:
I mean hm that statement deletes buttons if it exists and showButtons is false

Reply:
But I get this error (with `=== 0`)
```This condition will always return 'false' since the types '1 | 2' and '0' have no overlap.```

Reply:
But why you check length then it doesn't make any sense

Reply:
then just check if buttons isn't undefined kek

Reply:
At first, I had in the presenceData `buttons: []` so I had to delete it otherwise the presence didn't work properly

Reply:
Is 8bf7436 fine?